From d653dee28926c3f3617f61142d92494910d01742 Mon Sep 17 00:00:00 2001
From: Atsushi Watanabe <atsushi.w@ieee.org>
Date: Tue, 19 May 2026 12:16:28 +0900
Subject: [PATCH] Remove wrong tcflush call

---
 include/ypspur/serial.h |  1 -
 src/serial.c            | 15 +++------------
 2 files changed, 3 insertions(+), 13 deletions(-)

diff --git a/include/ypspur/serial.h b/include/ypspur/serial.h
index 13f0814..a135acb 100644
--- a/include/ypspur/serial.h
+++ b/include/ypspur/serial.h
@@ -40,7 +40,6 @@ int serial_close(void);
 int serial_write(char* buf, int len);
 int serial_recieve(int (*serial_event)(char*, int, double, void*), void* data);
 void serial_flush_in(void);
-void serial_flush_out(void);
 int encode_write(char* data, int len);
 
 #endif  // YPSPUR_SERIAL_H
diff --git a/src/serial.c b/src/serial.c
index 6aa5751..16d57ef 100644
--- a/src/serial.c
+++ b/src/serial.c
@@ -377,6 +377,9 @@ int serial_close(void)
   return 1;
 }
 
+/**
+ * @brief 可能なら入力バッファを破棄する
+ */
 void serial_flush_in(void)
 {
 #if !defined(__MINGW32__)
@@ -412,16 +415,6 @@ void serial_flush_in(void)
 #endif  // !defined(__MINGW32__)
 }
 
-void serial_flush_out(void)
-{
-#if !defined(__MINGW32__)
-  // Unix用
-  tcflush(g_device_port, TCOFLUSH);
-#else
-// Windows用
-#endif  // !defined(__MINGW32__)
-}
-
 // シリアルポートからの受信処理
 int serial_recieve(int (*serial_event)(char*, int, double, void*), void* data)
 {
@@ -533,7 +526,6 @@ int encode_write(char* data, int len)
   {
     return -1;
   }
-  serial_flush_out();
 
   return 0;
 }
@@ -552,7 +544,6 @@ int encode_int_write(char* data, int len)
   {
     return -1;
   }
-  serial_flush_out();
 
   return 0;
 }
